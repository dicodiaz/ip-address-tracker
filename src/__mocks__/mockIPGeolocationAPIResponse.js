const mockIPGeolocationAPIResponse = {
  ip: '8.8.8.8',
  location: {
    country: 'US',
    region: 'California',
    city: 'Mountain View',
    lat: 37.40599,
    lng: -122.078514,
    postalCode: '94043',
    timezone: '-07:00',
    geonameId: 5375481,
  },
  domains: ['0d2.net', '003725.com', '0f6.b0094c.cn', '007515.com', '0guhi.jocose.cn'],
  as: {
    asn: 15169,
    name: 'Google LLC',
    route: '8.8.8.0/24',
    domain: 'https://about.google/intl/en/',
    type: 'Content',
  },
  isp: 'Google LLC',
};

export const mockIPGeolocationAPIResponseWithDifferentIP = {
  ip: '1.1.1.1',
  location: {
    country: 'US',
    region: 'California',
    city: 'Los Angeles',
    lat: 34.05223,
    lng: -118.24368,
    postalCode: '90001',
    timezone: '-07:00',
    geonameId: 5368361,
  },
  domains: ['28xfy.cc', '7fnsj.cc', '850606.xyz', '98.ws', 'atlasreplymonitoring.it'],
  as: {
    asn: 13335,
    name: 'CLOUDFLARENET',
    route: '1.1.1.0/24',
    domain: 'https://www.cloudflare.com',
    type: 'Content',
  },
  isp: 'Cloudflare, Inc.',
};

export default mockIPGeolocationAPIResponse;
