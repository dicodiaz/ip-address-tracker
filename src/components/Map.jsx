import L from 'leaflet';
import PropTypes from 'prop-types';
import { Container } from 'react-bootstrap';
import { MapContainer, Marker, TileLayer } from 'react-leaflet';
import iconLocation from '../assets/icon-location.svg';

const markerIcon = L.icon({
  iconUrl: iconLocation,
});

const Map = ({ position, setMap }) => {
  return (
    <Container className="px-0" as="section" fluid>
      <MapContainer
        className="h-map"
        center={position}
        zoom={13}
        scrollWheelZoom={false}
        ref={setMap}
      >
        <TileLayer
          attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />
        <Marker position={position} icon={markerIcon} />
      </MapContainer>
    </Container>
  );
};

Map.propTypes = {
  position: PropTypes.arrayOf(PropTypes.number).isRequired,
  setMap: PropTypes.func.isRequired,
};

export default Map;
