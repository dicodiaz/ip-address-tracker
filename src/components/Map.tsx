import L, { LatLngExpression, Map as MapType } from 'leaflet';
import { FC, useEffect, useState } from 'react';
import { Container, Spinner } from 'react-bootstrap';
import { MapContainer, Marker, TileLayer } from 'react-leaflet';
import iconLocation from '../assets/icon-location.svg';
import { useAppSelector } from '../hooks/redux-hooks';
import { selectGeolocation } from '../redux/slices/geolocationSlice';

const markerIcon = L.icon({
  iconUrl: iconLocation,
});

const Map: FC = () => {
  const [map, setMap] = useState<MapType | null>(null);
  const geolocation = useAppSelector(selectGeolocation);
  const { lat, lng } = geolocation?.location ?? {};

  useEffect(() => {
    if (lat && lng) {
      map?.flyTo([lat, lng], map.getZoom());
    }
  }, [map, lat, lng]);

  if (!lat || !lng) {
    return (
      <Container
        as="section"
        className="vh-map d-flex justify-content-center align-items-center bg-secondary"
        fluid
      >
        <Spinner animation="border" role="status">
          <span className="visually-hidden">Loading...</span>
        </Spinner>
      </Container>
    );
  }

  const position: LatLngExpression = [lat, lng];

  return (
    <Container as="section" className="px-0" fluid>
      <MapContainer
        className="vh-map"
        center={position}
        zoom={13}
        scrollWheelZoom={false}
        ref={setMap}
      >
        <TileLayer
          attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />
        <Marker position={position} icon={markerIcon} />
      </MapContainer>
    </Container>
  );
};

export default Map;
