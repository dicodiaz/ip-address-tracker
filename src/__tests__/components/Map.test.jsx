import Map from '../../components/Map';
import mockIPGeolocationAPIResponse from '../../__mocks__/mockIPGeolocationAPIResponse';
import { renderWithProviders } from '../test-utils';

describe('Map component', () => {
  it('should render as expected', () => {
    const preloadedState = { geolocationReducer: { geolocation: mockIPGeolocationAPIResponse } };
    const { container } = renderWithProviders(<Map />, { preloadedState });
    expect(container).toMatchSnapshot();
  });
});
