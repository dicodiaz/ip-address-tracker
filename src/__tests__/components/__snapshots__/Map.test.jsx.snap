// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Map component > should render as expected 1`] = `
<div>
  <section
    class="px-0 container-fluid"
  >
    <div
      class="h-map leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
      style="position: relative;"
      tabindex="0"
    >
      <div
        class="leaflet-pane leaflet-map-pane"
        style="left: 0px; top: 0px;"
      >
        <div
          class="leaflet-pane leaflet-tile-pane"
        >
          <div
            class="leaflet-layer "
            style="z-index: 1;"
          >
            <div
              class="leaflet-tile-container leaflet-zoom-animated"
              style="z-index: 18; left: 0px; top: 0px;"
            >
              <img
                alt=""
                class="leaflet-tile"
                src="https://a.tile.openstreetmap.org/13/1318/3176.png"
                style="width: 256px; height: 256px; left: -9px; top: -249px;"
              />
            </div>
          </div>
        </div>
        <div
          class="leaflet-pane leaflet-overlay-pane"
        />
        <div
          class="leaflet-pane leaflet-shadow-pane"
        />
        <div
          class="leaflet-pane leaflet-marker-pane"
        >
          <img
            alt="Marker"
            class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
            role="button"
            src="/src/assets/icon-location.svg"
            style="left: 0px; top: 0px; z-index: 0; z-index: 0;"
            tabindex="0"
          />
        </div>
        <div
          class="leaflet-pane leaflet-tooltip-pane"
        />
        <div
          class="leaflet-pane leaflet-popup-pane"
        />
      </div>
      <div
        class="leaflet-control-container"
      >
        <div
          class="leaflet-top leaflet-left"
        >
          <div
            class="leaflet-control-zoom leaflet-bar leaflet-control"
          >
            <a
              aria-disabled="false"
              aria-label="Zoom in"
              class="leaflet-control-zoom-in"
              href="#"
              role="button"
              title="Zoom in"
            >
              <span
                aria-hidden="true"
              >
                +
              </span>
            </a>
            <a
              aria-disabled="false"
              aria-label="Zoom out"
              class="leaflet-control-zoom-out"
              href="#"
              role="button"
              title="Zoom out"
            >
              <span
                aria-hidden="true"
              >
                −
              </span>
            </a>
          </div>
        </div>
        <div
          class="leaflet-top leaflet-right"
        />
        <div
          class="leaflet-bottom leaflet-left"
        />
        <div
          class="leaflet-bottom leaflet-right"
        >
          <div
            class="leaflet-control-attribution leaflet-control"
          >
            <a
              href="https://leafletjs.com"
              title="A JavaScript library for interactive maps"
            >
              Leaflet
            </a>
             
            <span
              aria-hidden="true"
            >
              |
            </span>
             © 
            <a
              href="https://www.openstreetmap.org/copyright"
            >
              OpenStreetMap
            </a>
             contributors
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
